bindings/go/oci/blob/artifact_blob.go

```diff
@@ -11,28 +11,38 @@ import (
 	internaldigest "ocm.software/open-component-model/bindings/go/oci/internal/digest"
 )
 
-// ResourceBlob represents a blob of data that is associated with an OCM resource.
+// ArtifactBlob represents a blob of data that is associated with an OCM Source or Resource .
 // It implements various interfaces to provide blob-related functionality like
 // reading data, getting size, digest, and media type. This type is particularly
 // useful when working with OCI (Open Container Initiative) artifacts in the OCM
 // context, as it bridges the gap between OCM resources and OCI blobs.
-type ResourceBlob struct {
+type ArtifactBlob struct {
 	blob.ReadOnlyBlob
-	*descriptor.Resource
+	descriptor.Artifact
 	mediaType string
+	size      int64
 }
 
-// NewResourceBlobWithMediaType creates a new ResourceBlob instance with the given resource,
+// NewArtifactBlobWithMediaType creates a new ArtifactBlob instance with the given artifact,
 // blob data, and media type. This constructor ensures that all necessary
-// information is properly initialized for the ResourceBlob to function correctly.
-func NewResourceBlobWithMediaType(resource *descriptor.Resource, b blob.ReadOnlyBlob, mediaType string) (*ResourceBlob, error) {
+// information is properly initialized for the ArtifactBlob to function correctly.
+func NewArtifactBlobWithMediaType(artifact descriptor.Artifact, b blob.ReadOnlyBlob, mediaType string) (*ArtifactBlob, error) {
```

Comment:
Please find a way to not do this. Is this altering the `artifact` in the process of creating a new ArtifactBlob?

Because it's also setting the size of it. After that, `resource` is never used again anywhere.

Reply:
I agree, I changed the contract of the method to no longer modify the artifact and splitted the artifact modification into a different method.